docs/addons/jwt.md

```diff
@@ -0,0 +1,315 @@
+
+Configure **app/Config/AuthJWT.php** for your needs.
+
+### Set the Default Claims
+
+> **Note**
+> A payload contains the actual data being transmitted, such as user ID, role,
+> or expiration time. Items in a payload is called *claims*.
+
+Set the default payload items to the property `$defaultClaims`.
+
+E.g.:
+```php
+    public array $defaultClaims = [
+        'iss' => 'https://codeigniter.com/',
+    ];
+```
+
+The default claims will be included in all tokens issued by Shield.
+
+### Set Secret Key
+
+Set your secret key in the `$keys` property, or set it in your `.env` file.
+
+E.g.:
+```dotenv
+authjwt.keys.default.0.secret = 8XBFsF6HThIa7OV/bSynahEch+WbKrGcuiJVYPiwqPE=
+```
+
+It needs at least 256 bits random string. The length of the secret depends on the 
+algorithm we use. The default one is `HS256`, so to ensure that the hash value is 
+secure and not easily guessable, the secret key should be at least as long as the 
+hash function's output - 256 bits (32 bytes). You can get a secure random string 
+with the following command:
+
+```console
+php -r 'echo base64_encode(random_bytes(32));'
+```
+
+> **Note**
+> The secret key is used for signing and validating tokens.
+
+## Issuing JWTs
+
+To use JWT Authentication, you need a controller that issues JWTs.
+
+Here is a sample controller. When a client posts valid credentials (email/password),
+it returns a new JWT.
+
+```php
+// Routes.php
+$routes->post('auth/jwt', '\App\Controllers\Auth\LoginController::jwtLogin');
+
+// LoginController.php
+namespace App\Controllers\Auth;
+
+use App\Controllers\BaseController;
+use CodeIgniter\API\ResponseTrait;
+use CodeIgniter\HTTP\ResponseInterface;
+use CodeIgniter\Shield\Authentication\Authenticators\Session;
+use CodeIgniter\Shield\Authentication\JWTManager;
+use CodeIgniter\Shield\Authentication\Passwords;
+use CodeIgniter\Shield\Config\AuthSession;
+
+class LoginController extends BaseController
+{
+    use ResponseTrait;
+
+    /**
+     * Authenticate Existing User and Issue JWT.
+     */
+    public function jwtLogin(): ResponseInterface
+    {
+        // Get the validation rules
+        $rules = $this->getValidationRules();
+
+        // Validate credentials
+        if (! $this->validateData($this->request->getPost(), $rules)) {
+            return $this->failValidationErrors($this->validator->getErrors(), 422);
+        }
+
+        // Get the credentials for login
+        $credentials             = $this->request->getPost(setting('Auth.validFields'));
+        $credentials             = array_filter($credentials);
+        $credentials['password'] = $this->request->getPost('password');
+
+        /** @var Session $authenticator */
+        $authenticator = auth('session')->getAuthenticator();
+
+        // Check the credentials
+        $result = $authenticator->check($credentials);
+        if (! $result->isOK()) {
+            return $this->failUnauthorized($result->reason());
+        }
+
+        // Login is successful.
```

Comment:
The successful/unsuccessful login of the user must be recorded in the database. Even if it is only to receive a token, in any case, the user will login first with credentials. Also, pay attention to this issue, we cannot use filter `session` /`tokens` on **auth/jwt**.

Reply:
If we use `$authenticator->attempt($credentials)`, the login attempt will be stored in the database.
But it adds auth data in the session. The client is actually logged in with the Session Authenticator.

I thought the client does not need to be logged in actually, the current code just checks the credentials.

But I found that even if the user is banned, a new JWT will be issued.